src/main/webapp/app/quiz/edit/multiple-choice-question/edit-multiple-choice-question.component.ts

```diff
@@ -97,114 +71,101 @@ export class EditMultipleChoiceQuestionComponent implements OnInit, OnChanges, A
-        this.questionEditor.getEditor().focus();
-        this.questionEditor.getEditor().clearSelection();
-        const lines = this.questionEditorText.split('\n').length;
-        const range = this.questionEditor.getEditor().selection.getRange();
-        this.questionEditor.getEditor().moveCursorTo(this.questionEditor.getEditor().getCursorPosition().row, Number.POSITIVE_INFINITY);
-        this.questionEditor.getEditor().insert(textToAdd);
-        range.setStart(lines, 4);
-        range.setEnd(lines, textToAdd.length - 1);
-        this.questionEditor.getEditor().selection.setRange(range, false);
+    prepareForSave(): void {
+        this.cleanupQuestion();
+        this.markdownEditor.parse();
+        this.questionUpdated.emit();
     }
 
     /**
-     * @function addHintAtCursor
-     * @desc Adds the markdown for a hint at the current cursor location
+     * @function cleanupQuestion
+     * @desc Clear the question to avoid double assignments of one attribute
      */
-    addHintAtCursor(): void {
-        this.artemisMarkdown.addHintAtCursor(this.questionEditor.getEditor());
+    private cleanupQuestion() {
+        // Reset Question Object
+        this.question.answerOptions = [];
+        this.question.text = null;
+        this.question.explanation = null;
+        this.question.hint = null;
+        this.question.hasCorrectOption = null;
     }
 
     /**
-     * This function opens the modal for the help dialog.
+     * @function domainCommandsFound
+     * @desc 1. Gets the {array} containing the textLine with the domainCommandIdentifier and creates a new multiple choice question
+     *       by assigning the textLines according to the domainCommandIdentifiers to the multiple choice question attributes.
+     *       (question text, explanation, hint, answeroption(correct/wrong)
+     *       2. For each answer option a new AnswerOption entity is created by assiging the answeroption text and setting
+     *       the answeroption correct/wrong
+     *       3.Important is that the resetMultipleChoicePreview() is triggered to notify the parent component
+     *       about the changes within the question and to trigger the checking method for correct values
+     * @param {array} of markdownTextLine {string} with the corresponding domainCommand {DomainCommand} identifier
      */
-    open(content: any) {
-        this.modalService.open(content, { size: 'lg' });
-    }
+    domainCommandsFound(domainCommands: [string, DomainCommand][]): void {
+        this.cleanupQuestion();
+        let currentAnswerOption = new AnswerOption();
 
-    /**
-     * @function addExplanationAtCursor
-     * @desc Adds the markdown for an explanation at the current cursor location
-     */
-    addExplanationAtCursor(): void {
-        this.artemisMarkdown.addExplanationAtCursor(this.questionEditor.getEditor());
+        for (const [textLine, command] of domainCommands)  {
```

Comment:
as far as I understand `textLine` can also consist of multiple lines, so please refactor it into `text` instead